integrations/ultralytics-yolov5/tutorials/sparsifying_yolov5_using_recipes.md

```diff
@@ -0,0 +1,230 @@
+    - If your hardware does support quantized networks (VNNI instruction set on CPUs, for example), we recommend using the `pruned quantized` recipe. The recipe to use depends on how long you are willing to train and how crucial full recovery is. Consult the table for this comparison.
+    - When running quantized models, the memory footprint for training will significantly increase (roughly 3x). It is recommended to train at a high batch size at first. This will fail with an out-of-memory exception once quantization starts. Once this happens, use the `last.pt` weights from that run to resume training with a lower batch size.
+
+3. To begin applying one of the recipes, use the `--recipe` argument within the Ultralytics [train script](https://github.com/neuralmagic/yolov5/blob/master/train.py).
+   The recipe argument is combined with our previous training command and COCO pre-trained weights to run the recipes over the model. For example, a command for YOLOv5s would look like this:
+   ```bash
+   python train.py --cfg ../models/yolov5s.yaml --weights PATH_TO_COCO_PRETRAINED_WEIGHTS --data coco.yaml --hyp data/hyp.scratch.yaml --recipe PATH_TO_SPARSIFICATION_RECIPE
+   ```
+    After applying a recipe, you are ready to export for inference.
+
+## Exporting for Inference
+
+This step loads a checkpoint file of the best weights measured on the validation set, and converts it into the more common inference formats. 
+Then, you can run the file through a compression algorithm to reduce its deployment size and run it in an inference engine such as [DeepSparse](https://github.com/neuralmagic/deepsparse).
+
+When you applied a recipe in the previous step, the sparsification run created a new `exp` directory under the yolov5 runs directory:
+
+```
+|-- VOC
+|-- data
+|-- models
+|-- recipes
+|-- tutorials
+|-- yolov5
+|   |-- data
+|   |-- models
+|   |-- runs
+|   |   `-- train
+|   |       |-- exp
+|   |       |-- exp1
+|   |       |   |-- weights
+|   |       |   |   |-- best.pt
+|   |       |   |   `-- last.pt
+|   |       |   |-- F1_curve.png
+|   |       |   |-- PR_curve.png
+|   |       |   |-- P_curve.png
+|   |       |   |-- R_curve.png
+|   |       |   |-- confusion_matrix.png
+|   |       |   `-- ...
+|   |-- train.py
+|   `-- ...
+|-- README.md
+`-- setup_integration.sh
+```
+
+The `best.pt` file contains a checkpoint of the best weights measured on the validation set.
+These weights can be loaded into the `train.py` and `test.py` scripts now. 
+However, other formats are generally more friendly for other inference deployment platforms, such as [ONNX](https://onnx.ai/).
+
+The [`export.py` script](https://github.com/neuralmagic/yolov5/blob/master/models/export.py) handles the logic behind loading the checkpoint and converting it into the more common inference formats, as described here.
+
+1. Enter the following command to load the PyTorch graph, convert to ONNX, and then correct any misformatted pieces for the pruned and quantized models.
```

Comment:
Will it be obvious to the user what "misformatted pieces" are? Should you provide an examples clause....e.g., such as x, y, or z? Or not needed and it will be quite obvious?

Reply:
I think It should be pretty obvious, will let @markurtz take the call

Reply:
This is supposed to be stating that any misformatted pieces for the graph will be corrected by the command. I'll make a few edits to make this more obvious